File: battleship-game/src/game/ship.ts
export class Ship {
    length: number;
    coordinates: { x: number, y: number }[];
    hits: Set<string>;

    constructor(length: number, coordinates: { x: number, y: number }[]) {
        this.length = length;
        this.coordinates = coordinates;
        this.hits = new Set();
    }

    hit(x: number, y: number): boolean {
        const coord = `${x},${y}`;
        if (this.coordinates.some(c => c.x === x && c.y === y)) {
            this.hits.add(coord);
            return true;
        }
        return false;
    }

    isSunk(): boolean {
        return this.hits.size === this.length;
    }
}

export default Ship;

File: src/game/ship.py
class Ship:
    def __init__(self, length, coordinates):
        self.length = length
        self.coordinates = coordinates
        self.hits = set()

    def hit(self, x, y):
        coord = (x, y)
        if coord in self.coordinates:
            self.hits.add(coord)
            return True
        return False

    def is_sunk(self):
        return len(self.hits) == self.length